src/components/[guild]/DiscordBotPermissionsChecker.tsx

```diff
@@ -0,0 +1,228 @@
+        Seems like you've changed our bot role's name on your Discord server. It
+        might not work properly unless you keep its original name (
+        <i>Guild.xyz bot</i>).
+      </Text>
+    ),
+  },
+}
+
+const DiscordBotPermissionsChecker = () => {
+  const { isAdmin } = useGuildPermission()
+  const { id, guildPlatforms, roles } = useGuild()
+  const discordRewards =
+    guildPlatforms?.filter((gp) => gp.platformId === PlatformType.DISCORD) ?? []
+  const discordRewardIds = discordRewards.map((gp) => gp.id)
+  const relevantDiscordRoles =
+    roles
+      ?.filter((role) =>
+        role.rolePlatforms.some((rp) =>
+          discordRewardIds.includes(rp.guildPlatformId)
+        )
+      )
+      .flatMap((role) => role.rolePlatforms)
+      .map((rp) => rp.platformRoleId) ?? []
+
+  const [errorType, setErrorType] = useState<PermissionModalType>()
+  const { isOpen, onOpen, onClose } = useDisclosure()
+
+  const fetchDiscordPermissions = () =>
+    Promise.all(
+      discordRewards?.map((gp) =>
+        fetcher(`/discord/permissions/${gp.platformGuildId}`)
+      )
+    )
+
+  const { data } = useSWRImmutable<DiscordPermissions[]>(
+    discordRewards.length > 0 && isAdmin ? ["discordPermissions", id] : null,
+    fetchDiscordPermissions
+  )
+
+  const toastWithButton = useToastWithButton()
+  const toastIdRef = useRef<ToastId>()
+  const toastOptions: ActionToastOptions = {
+    status: "warning",
+    duration: null,
+    isClosable: false,
+    buttonProps: {
+      children: "Fix problem",
+      rightIcon: <Wrench />,
+      onClick: onOpen,
+    },
+    secondButtonProps: {
+      children: "Later",
+      variant: "ghost",
+    },
+  }
+
+  useEffect(() => {
+    if (!data || !!toastIdRef.current) return
+
+    for (const [index, permissionInfo] of data.entries()) {
+      const serverName = discordRewards[index].platformGuildName
+
+      const permissionsNotGranted = Object.values(permissionInfo.permissions).filter(
+        (perm) => !perm.value && perm.name !== "Administrator"
+      )
+
+      if (permissionsNotGranted.length > 0) {
+        toastIdRef.current = toastWithButton({
+          title: "Missing permissions",
+          description: `We've noticed that the Guild.xyz bot is missing the following permissions on the ${serverName} Discord server: ${permissionsNotGranted
+            .map((perm) => PRETTY_PERMISSION_NAMES[perm.name] ?? perm.name)
+            .join(", ")}`,
+          ...toastOptions,
+        })
+        setErrorType("PERMISSIONS")
+        return
+      }
+
+      const discordBotsRole = permissionInfo.roleOrders.find(
+        (role) => role.roleName === GUILD_BOT_ROLE_NAME
+      )
+
+      if (!discordBotsRole) {
+        toastIdRef.current = toastWithButton({
+          title: "Guild.xyz Discord bot is misconfigured",
+          description: `Seems like you've changed our bot's role name on the ${serverName} Discord server. It might not work properly unless you keep its original name.`,
```

Comment:
The truth is that if the user renames both the `Guild.xyz bot` role and the changes the bot's nickname, it'll work just fine (e.g. renames the both the role and the bot to `My fav bot`, it'll work fine - more info in the Discord thread). But, we can't really check the bot's nickname right now, so I think it's better to just show this error message and encourage users to keep the bot's original name...